### PR TITLE
Use vite for Storybook

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

@benchristel got a custom dev server up and running using Vite. This got me thinking on Friday if we could use Vite for Storybook. Turns out, we can!

This results in much faster loading times in Storybook, plus the Storybook build step that is used to publish new versions to `khan.github.io/perseus` are much faster also.

image

# Performance 

Also, building storybook for `khan.github.io/perseus` is much faster! 

## Branch: `main` 

```sh
$ hyperfine -i "yarn build-storybook"
Benchmark 1: yarn build-storybook
  Time (mean ± σ):     19.682 s ±  0.278 s    [User: 57.151 s, System: 2.995 s]
  Range (min … max):   19.269 s … 20.252 s    10 runs
```

## Branch: `jer/storybook-vite`

```sh
$ hyperfine -i "yarn build-storybook"
Benchmark 1: yarn build-storybook
  Time (mean ± σ):     10.442 s ±  0.273 s    [User: 16.751 s, System: 1.782 s]
  Range (min … max):   10.114 s … 10.921 s    10 runs
 
  Warning: Ignoring non-zero exit code.
```

Issue: "none"

## Test plan:

`yarn build-storybook` succeeds
`yarn start` -> navigate around -> :thumbsup:

### CODE REVIEW

File: .storybook/preview.tsx
Comment (jeremywiebe) [inline .storybook/preview.tsx:1]: This rename was required by the changes in this PR. We _are_ using JSX in this file so it should be a `.tsx` anyways.

File: dev/vite.config.ts [hunk 1]
```diff
@@ -7,16 +7,19 @@ import {defineConfig} from "vite";
 // Create aliases for each package in the Perseus monorepo, so Vite knows
 // where to look when a file imports e.g. @khanacademy/perseus.
 const packageAliases = {};
-glob.sync(resolve("../packages/*/package.json")).forEach((packageJsonPath) => {
```
Comment (jeremywiebe) [inline dev/vite.config.ts:10]: I had to change this to use a base path to resolve relative to now that I'm importing it in the Storybook config.

File: packages/math-input/src/components/keypad/keypad-pages/keypad-pages.stories.tsx [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import {action} from "@storybook/addon-actions";
-import {withKnobs} from "@storybook/addon-knobs";
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/keypad-pages/keypad-pages.stories.tsx:2]: `@storybook/addon-knobs` is deprecated.  These stories don't look like they were working properly anyways as I couldn't see any controls available in the existing, published stories: https://khan.github.io/perseus/?path=/story/math-input-components-keypad-pages--numeric-input

File: vendor/jsdiff/jsdiff.js
Comment (jeremywiebe) [inline vendor/jsdiff/jsdiff.js:1]: @benchristel already migrated the other two libraries in this folder to be standard ES6 modules. Vite was having issues with this one so I did the same. Best viewed by turning off whitespace differences. 

image

File: package.json [hunk 3]
```diff
@@ -42,12 +42,12 @@
     "@sheerun/mutationobserver-shim": "^0.3.3",
     "@storybook/addon-a11y": "^7.6.17",
     "@storybook/addon-actions": "^7.6.17",
+    "@storybook/addon-controls": "^7.6.17",
     "@storybook/addon-essentials": "^7.6.17",
-    "@storybook/addon-knobs": "^7.0.2",
```
Comment (jeremywiebe) [inline package.json:46]: This library is deprecated and no longer supported by Storybook.

**Comment by jeremywiebe** [comment]
Tagging in @jandrade @jeresig @kevinbarabash  in case you're interested. :)

File: .changeset/strong-guests-wave.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@khanacademy/math-input": patch
+---
+
+Fix handling of "Fractions" keypad in the keypad tab bar icon component
```
Comment (nedredmond) [inline .changeset/strong-guests-wave.md:5]: Is this supposed to be in the same PR?

**Review by nedredmond — APPROVED** [review]
I love this

**Comment by jeresig** [comment]
Very nice!

**Pull request merged**